Cargo.toml

```diff
@@ -21,9 +21,41 @@ clippy.missing_const_for_fn = "deny"
 clippy.too_long_first_doc_paragraph = "warn"
 
 [workspace.dependencies]
-
-[profile.test.package.proptest]
-opt-level = 3
+async-trait = "0.1.83"
```

Comment:
I'd really like to remove this dependency. It is a legacy thing that we used to use because we did not have Rust 1.75

See: https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits/

Reply:
What if we annotate the result with lifetimes and/or add `Sync` bounds to some of the arguments?
We are in the midst of removing all `async_trait` macros and dependency in our codebases at Alpen.
I don't want `strata-p2p` to be the ugly duck.